ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -403,6 +454,38 @@ func (r *RayDashboardClient) SubmitJob(ctx context.Context, rayJob *rayv1.RayJob
 	return jobResp.JobId, nil
 }
 
+// Get Job Log
+func (r *RayDashboardClient) GetJobLog(ctx context.Context, jobName string, log *logr.Logger) (*string, error) {
+	log.Info("Get ray job log", "rayJob", jobName)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, r.dashboardURL+JobPath+jobName+"/logs", nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := r.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
```

Comment:
Question: Here and in general, is it better to raise an error if the job is not found?

Reply:
Fixed